examples/COUPLE/simple/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.10)
+project(simple CXX)
+
+set(LAMMPS_SRC_DIRECTORY "" CACHE PATH "Path for lammps source")
+if(NOT LAMMPS_SRC_DIRECTORY STREQUAL "" AND EXISTS ${LAMMPS_SRC_DIRECTORY}/cmake/CMakeLists.txt)
+  set(BUILD_LIB ON)
+  execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${LAMMPS_SRC_DIRECTORY}/src ${LAMMPS_SRC_DIRECTORY}/src/lammps)
```

Comment:
@akohlmey this is a hack to generate `lammps/` prefix for the header, do you have a better solution?

Reply:
This is a bit of a moving target, since we need to have a discussion with the other LAMMPS core developers first about which part of the C++ API we want to expose. With the library interface the situation is trivial, since everything needs to be wrapped, but with the C++ API this is not as clear. In the past this has been mostly side-stepped by telling people "just look at library.cpp". I would like to formalize this and then have the "public" C++ APIs properly documented through doxygen and this added to a programmer's guide. Most of the required information is already there as comments in the source code, it just needs to be put in the right places and formatted so that it can be imported into the manual.

Reply:
Then let's leave the hack there for now and remove it once we decided.